CDCgov/prime-reportstream
# 15153: Implement new events pattern

This PR implements the new events as defined in the SRD

Test Steps:

Overall, tester should look at events defined [here](https://lucidgov.app/lucidchart/b2251f9d-ec5f-4186-9a87-cd2443ab0000/edit?viewport_loc=-9761,-5997,20162,14599,0_0&invitationId=inv_a428973e-8d90-429d-b61f-9305464f48f9) (V0.2 version!!!) and confirm that the contents of the specifics have the right params.  The few differences are:

- filesize is not going to be on any event other than report received
- reportId -> childReportId

Confirm via logging:
1. Turn on prime-router
1. Run the smoketest locally
1. Confirm in the logs that you see all the expected events

Confirm via application insights
1. Follow the setup for local connection to app insights in the TEST env in `observability/local-dev-setup.md`
1. Turn on prime-router
1. Run the smoketest locally
1. Confirm you see the expected events in app insights

## Changes
- Adds a new DSL for creating and sending events
- Updates the pipeline to send events as defined in the SRD
- Fixes a bug with the action log field names

## Checklist

### Testing

### Process

## Linked Issues
- #15153 

## To Be Done
- https://github.com/CDCgov/prime-reportstream/issues/15337

## Specific Security-related subjects a reviewer should pay specific attention to
